Source/WebCore/fileapi/Blob.cpp

```diff
@@ -183,6 +204,7 @@ Blob::Blob(DeserializationContructor, ScriptExecutionContext* context, const URL
 Blob::Blob(ScriptExecutionContext* context, const URL& srcURL, long long start, long long end, const String& type)
     : ActiveDOMObject(context)
     , m_type(normalizedContentType(type))
+    , m_memoryCost(end - start)
```

Comment:
This is probably incorrect in case end is for instance -1, or if slicing a File.
It might be better to pass a memoryCost parameter and compute its value in Blob::slice.